### DISCUSSION TITLE
Lesson 8 - withdraw function not working.

### DISCUSSION THREAD

**Discussion by solidityenj0yer** [Cyfrin/foundry-full-course-cu#2026] [Q&A]
Hello everyone, I'm having trouble withdrawing the deposited currency from the local host website. Even thought, I am connected to the account that I keep depositing from, it's not letting me withdraw. 

Here is the console error that I'm getting:
![error](https://github.com/user-attachments/assets/c8edb5ed-1d9a-42fe-9953-98910524b212)

The code for the Withdraw function:
![withdraw](https://github.com/user-attachments/assets/361c21bd-0b00-45fc-991b-c17d7e4291d0)

Do you have any suggestions? I get that it's probably not that important but I just wanted to see why I'm getting the error.

**Comment by EngrPips** [comment]
Hello @solidityenj0yer, do you have a restriction on who can withdraw the funds in your contract?

**Reply by solidityenj0yer** [reply]
I am not sure of that, because this lesson I was just supposed to clone the code from Patrick's depository - https://github.com/Cyfrin/html-fund-me-cu , its also JavaScript and I can not really read it, Patrick mentions that it's only the owner that can use the Withdraw function.

**Reply by EngrPips** [reply]
Oh, I see. I wouldn't let that block me tho. Since I can't even see the contract, I am not sure what went wrong, but the error you showed indicated that the transaction got reverted with a custom error. I believe the main goal of that lesson is to learn how to do some front-end stuff, so if you think you achieve that you should move on.

**Reply by solidityenj0yer** [reply]
Yeah, I figured. thanks for the insight!

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
